www/js/lib/uiUtil.js

```diff
@@ -84,6 +84,30 @@ define([], function() {
         return url.replace(regexpRemoveUrlParameters, "$1");
     }
 
+    /**
+     * Derives the URL.pathname from a relative or semi-relative URL using the given base ZIM URL
+     * 
+     * @param {String} url The (URI-encoded) URL to convert (e.g. "Einstein", "../Einstein",
+     *      "../../I/im%C3%A1gen.png", "-/s/style.css", "/A/Einstein.html")
+     * @param {String} base The base ZIM URL of the currently loaded article (e.g. "A/" or "A/subdir1/subdir2/")
+     * @returns {String} The derived ZIM URL in decoded form (e.g. "A/Einstein", "I/imágen.png")
+     */
+    function deriveURL(url, base) {
```

Comment:
I think this function should be renamed to something more meaningful, like `constructZimUrlFromRelativeUrl`